source/features/enable-file-links-in-compare-view.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import GitBranchIcon from 'octicon/git-branch.svg';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import GitHubURL from '../github-helpers/github-url';
+import {getCurrentBranch} from '../github-helpers';
+
+function handlePRMenuOpening(event: delegate.Event): void {
+	event.delegateTarget.classList.add('rgh-actionable-link'); // Mark this as processed
+
+	// This solution accounts for:
+	// - Branches with slashes in it
+	// - PRs opened from the default branch
+	const dropdown = event.delegateTarget.nextElementSibling!;
+	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
+	// You can only get the user and repository, since PR's open from the default branch will not have the branch in the pathname.
+	const [user, repository] = select<HTMLAnchorElement>('.commit-ref.head-ref a')!.pathname.split('/');
+	const url = new GitHubURL(viewFile.href);
+	url.assign({
+		user,
+		repository,
+		branch: getCurrentBranch()
+	});
+
+	viewFile.href = String(url);
```

Comment:
As the code comment mention, this used to work. https://github.com/sindresorhus/refined-github/pull/3111 broke it and the previous version of the code is cleaner than the current one. Does this still work?

```suggestion
	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
	const pathParts = viewFile.pathname.split('/'); // Example pathname: $owner/$repository/blob/$sha/$path_to_file.tsx
	pathParts[4] = getCurrentBranch();
	viewFile.pathname = pathParts.join('/');
```

Reply:
No since the view file link is the local repo not the head repo. Hence my mistake that was testing on my own repo

The last fully working version is this.
https://github.com/yakov116/refined-github/blob/a24d9460d36e5147f5a9f389923f2eb75f34e34f/source/features/fix-view-file-link-in-pr.tsx#L25-L34

Reply:
I think the difference is that the link will still point to the base repo rather than the head repo, but it's not a big deal as long as it workks.